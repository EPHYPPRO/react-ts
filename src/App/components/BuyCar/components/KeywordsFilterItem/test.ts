import { render } from './test.utils';
import fake from 'casual';
import { fireEvent } from 'react-testing-library';

describe('KeywordsFilterItem', () => {
  it('should render properly', async () => {
    const { component } = render();

    expect(component).toMatchSnapshot();
  });

  it('should change store state', async () => {
    const fakeText = fake.title;
    const { component, getStoreState } = render();
    const input = component.querySelector('input');

    input && fireEvent.change(input, { target: { value: fakeText } });

    const state = getStoreState();

    expect(state.keywords).toBe(fakeText);
  });

});
