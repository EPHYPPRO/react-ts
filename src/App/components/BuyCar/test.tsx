import React from 'react';
import { BuyCar } from '.';
import { getConfiguredRender } from 'test-utils/configured-render';

const render = getConfiguredRender(<BuyCar />);

describe('BuyCar', () => {
  it('should render properly', async () => {
    const { container } = render();

    expect(container.firstChild).toMatchSnapshot();
  });
});
