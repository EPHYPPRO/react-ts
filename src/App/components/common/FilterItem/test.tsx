import * as React from 'react';
import { FilterItem, FilterItemProps } from '.';
import { getConfiguredRender } from 'test-utils/configured-render';
import fake from 'casual';

const render = getConfiguredRender<FilterItemProps>(<FilterItem />);

describe('FilterItem', () => {
  it('should render properly', async () => {
    const { component } = render({
      label: 'label text',
      children: 'content'
    });

    expect(component).toMatchSnapshot();
  });

  it('should render children and do not render label element', async () => {
    const children = fake.text;
    const { container, queryByTestId } = render({ children });

    expect(container).toHaveTextContent(new RegExp(`${children}`));
    expect(queryByTestId('label')).toBeFalsy();
  });

  it('should render label', async () => {
    const label = fake.title;
    const { getByTestId } = render({ label });

    expect(getByTestId('label')).toHaveTextContent(
      new RegExp(label)
    );
  });
});
