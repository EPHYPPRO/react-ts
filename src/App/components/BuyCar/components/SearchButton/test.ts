import { render, getOptions } from './test.utils';

describe('SearchButton', () => {
  it('should render properly', async () => {
    const { component } = render();

    expect(component).toMatchSnapshot();
  });

  it("should be disabled when at least one of ids doesn't picked", async () => {
    const { component, rerender } = render(
      {},
      getOptions(1, 1)
    );

    const searchButton = component.querySelector(
      'button'
    ) as HTMLButtonElement;

    expect(searchButton.disabled).toBeFalsy();

    const checkForDisabledState = (
      brandId: number,
      modelId: number
    ) => {
      rerender({}, getOptions(brandId, modelId));
      expect(searchButton.disabled).toBeTruthy();
    };

    checkForDisabledState(-1, 1);
    checkForDisabledState(1, -1);
    checkForDisabledState(-1, -1);
  });

  it('should call handler on button press', async () => {
    const onClick = jest.fn();
    const { getByText } = render(
      {
        onClick
      },
      getOptions(1, 1)
    );

    const searchButton = getByText(/Search/i);
    searchButton.click();

    expect(onClick).toHaveBeenCalled();
  });
});
