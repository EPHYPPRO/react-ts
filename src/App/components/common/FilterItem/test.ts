import fake from 'casual';
import { render, checkForLabelRender } from './test.utils';

describe('FilterItem', () => {
  it('should render properly', async () => {
    const { component } = render({
      label: 'label text',
      children: 'content'
    });

    expect(component).toMatchSnapshot();
  });

  it('should render children and do not render label element', async () => {
    const children = fake.text;
    const { container, queryByTestId } = render({ children });

    expect(container).toHaveTextContent(new RegExp(`${children}`));
    expect(queryByTestId('label')).toBeFalsy();
  });

  it('should render label', async () => {
    checkForLabelRender(render);
  });
});
