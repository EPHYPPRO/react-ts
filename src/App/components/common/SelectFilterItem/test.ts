import fake from 'casual';
import { buildItem, render } from './test.utils';
import { fireEvent } from 'react-testing-library';
import { checkForLabelRender } from '../FilterItem/test.utils';

describe('SelectFilterItem', () => {
  it('should render properly', async () => {
    const { component } = render({
      id: 1,
      items: [{ id: 1, text: 'item 1' }, { id: 2, text: 'item 2' }],
      handleChange: jest.fn(),
      defaultItemValue: 'item default',
      label: 'label text'
    });

    expect(component).toMatchSnapshot();
  });

  it('should pick default item', async () => {
    const defaultItemText = fake.title;
    const { queryByText } = render({
      defaultItemValue: defaultItemText
    });

    expect(queryByText(defaultItemText)).toBeTruthy();
  });

  it('should show linear loader when isLoading', async () => {
    const { rerender, queryByTestId } = render({
      isLoading: true
    });

    expect(queryByTestId('loader')).toBeTruthy();

    rerender({
      isLoading: false
    });

    expect(queryByTestId('loader')).toBeFalsy();
  });

  it('should show error when error occured', async () => {
    const { rerender, queryByTestId } = render({
      error: true
    });

    expect(queryByTestId('error-message')).toBeTruthy();

    rerender({
      error: false
    });

    expect(queryByTestId('error-message')).toBeFalsy();
  });

  it('should call handleChange callback', async () => {
    const item1 = buildItem();
    const item2 = buildItem();
    const handleChange = jest.fn();
    render({
      handleChange,
      items: [item1, item2],
    });

    // items block rendered in the end of the body
    const element1 = document.body.querySelector<HTMLElement>(
      'div.mui-fixed ul > li:nth-child(1)'
    );
    element1 && fireEvent.click(element1);

    expect(handleChange).toHaveBeenCalledWith(item1.id);
  });

  it('should render label', async () => {
    checkForLabelRender(render);
  });
});
