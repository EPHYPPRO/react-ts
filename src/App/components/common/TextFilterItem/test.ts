import fake from 'casual';
import { render } from './test.utils';
import { fireEvent } from 'react-testing-library';
import { checkForLabelRender } from '../FilterItem/test.utils';

describe('TextFilterItem', () => {
  it('should render properly', async () => {
    const { component } = render({
      handleChange: jest.fn(),
      label: 'label text',
      text: fake.title
    });

    expect(component).toMatchSnapshot();
  });

  it('should call handleChange callback', async () => {
    const fakeText = fake.title;
    const handleChange = jest.fn();

    const { component } = render({
      handleChange
    });

    const input = component.querySelector('input');

    input && fireEvent.change(input, { target: { value: fakeText } });

    expect(handleChange).toHaveBeenCalledWith(fakeText);
  });

  it('should render label', async () => {
    checkForLabelRender(render);
  });

});
