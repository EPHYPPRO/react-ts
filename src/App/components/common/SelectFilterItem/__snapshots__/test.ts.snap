// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SelectFilterItem should render properly 1`] = `
<div
  class="filter-item"
>
  <div
    class="label"
    data-testid="label"
  >
    label text
  </div>
  <div
    class="MuiInputBase-root-21 MuiInput-root-8 MuiInput-underline-12 MuiInputBase-fullWidth-30 MuiInput-fullWidth-15"
    data-testid="select"
  >
    <div
      class="MuiSelect-root-1"
    >
      <div
        class="MuiSelect-select-2 MuiSelect-selectMenu-5 MuiInputBase-input-31 MuiInput-input-16"
        aria-pressed="false"
        tabindex="0"
        role="button"
        aria-haspopup="true"
      >
        item 1
      </div>
      <input
        type="hidden"
        value="1"
      />
      <svg
        class="MuiSvgIcon-root-38 MuiSelect-icon-7"
        focusable="false"
        viewBox="0 0 24 24"
        aria-hidden="true"
        role="presentation"
      >
        <path
          d="M7 10l5 5 5-5z"
        />
      </svg>
    </div>
  </div>
</div>
`;
